core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -234,7 +234,7 @@ public void run() {
      * <p>
      * DO NOT USE THIS OUTSIDE OF A TEST
      */
-    void awaitWrittenForTest() throws InterruptedException {
+    protected void awaitWrittenForTest() throws InterruptedException {
```

Comment:
Im not keen on this. Maybe just ping pong request and set special header to attribute value?
https://github.com/undertow-io/undertow/blob/main/core/src/test/java/io/undertow/server/handlers/SetAttributeTestCase.java